tests/models/hydrology/test_hydrology_tools.py

```diff
@@ -31,80 +16,107 @@ def test_setup_hydrology_input_current_timestep(
     )
 
-    dummy_climate_data["soil_moisture"] = xr.concat(
-        [
-            DataArray(np.full((13, 3), np.nan), dims=["layers", "cell_id"]),
-            DataArray(np.full((2, 3), 50), dims=["layers", "cell_id"]),
-        ],
-        dim="layers",
-    )
+    layer_structure = fixture_core_components.layer_structure
     result = setup_hydrology_input_current_timestep(
         data=dummy_climate_data,
         time_index=0,
         days=30,
         seed=42,
-        layer_roles=fixture_core_components.layer_structure.layer_roles,
-        soil_layer_thickness=np.array([[10, 10, 10], [100, 100, 100]]),
+        layer_structure=layer_structure,
+        soil_layer_thickness_mm=layer_structure.soil_layer_thickness * 1000,
         soil_moisture_capacity=0.9,
         soil_moisture_residual=0.1,
         core_constants=CoreConsts(),
         latent_heat_vap_equ_factors=[1.91846e6, 33.91],
     )
 
-    # Check if all variables were created
+    # Check if all variables were created TODO switch back to subcanopy
     var_list = [
         "current_precipitation",
-        "subcanopy_temperature",
-        "subcanopy_humidity",
-        "subcanopy_pressure",
-        "subcanopy_wind_speed",
-        "leaf_area_index_sum"
-        "current_evapotranspiration"
-        "soil_layer_heights"
-        "soil_layer_thickness"
-        "top_soil_moisture_capacity_mm"
-        "top_soil_moisture_residual_mm"
-        "soil_moisture_true",
-        "previous_accumulated_runoff"
-        "previous_subsurface_flow_accumulated"
+        # "subcanopy_temperature",
+        # "subcanopy_humidity",
+        # "subcanopy_pressure",
+        # "subcanopy_wind_speed",
+        "surface_temperature",
+        "surface_humidity",
+        "surface_pressure",
+        "surface_wind_speed",
+        "leaf_area_index_sum",
+        "current_evapotranspiration",
+        # "soil_layer_heights",
+        # "soil_layer_thickness",
+        "top_soil_moisture_capacity_mm",
+        "top_soil_moisture_residual_mm",
+        # "soil_moisture_true",
+        "previous_accumulated_runoff",
+        "previous_subsurface_flow_accumulated",
         "groundwater_storage",
+        "soil_moisture_mm",
+        "latent_heat_vapourisation",
+        "molar_density_air",
     ]
 
-    variables = [var for var in result if var not in var_list]
-    assert variables
+    assert set(result.keys()) == set(var_list)
 
     # check if climate values are selected correctly
     np.testing.assert_allclose(
         np.sum(result["current_precipitation"], axis=1),
         (dummy_climate_data["precipitation"].isel(time_index=0)).to_numpy(),
     )
     np.testing.assert_allclose(
-        result["subcanopy_temperature"], dummy_climate_data["air_temperature"][11]
+        result["surface_temperature"],
+        dummy_climate_data["air_temperature"]
+        .isel(layers=fixture_core_components.layer_structure.role_indices["surface"])
+        .squeeze(),
     )
```

Comment:
This seems a bit odd to me, is there no more direct way to select a layer?

Reply:
Yup - good point.